### PR TITLE
Evil commentary contrib layer.

### PR DESCRIPTION
**Pull request by person808**
Supersedes #831.

### CODE REVIEW

**Comment by person808** [comment]
Removed use-package. It seems to have broken loading.

File: contrib/evil-commentary/README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Vim-empty-lines contribution layer for Spacemacs
```
Comment (trishume) [inline contrib/evil-commentary/README.md]: Forgot to change the heading

**Comment by trishume** [comment]
This looks cool, thanks for the PR!

`use-package` is currently undergoing a major upgrade with breaking changes, it's possible that the documentation doesn't currently line up with how it should be used with the version you are using.

Your layer should definitely use `use-package` but it might be a good idea to maybe wait a day or two and ask for help in the Gitter chat with the problem of it breaking loading.

File: contrib/evil-commentary/README.md (hunk 1 above)
Comment (person808) [inline contrib/evil-commentary/README.md]: Thanks. Fixed +1:

**Comment by person808** [comment]
Using ```:defer t``` forces lazy loading in use-package, correct? When I omit that line use-package works fine for evil-commentary.

**Comment by trishume** [comment]
@person808 yah, it does auto-guesstimation of how to lazy-load it. Some packages don't work well with the auto-guesstimation. In that case you can either omit `:defer t` or use some of the manual ways you can tell it how to lazy load like `:commands`. Either way works, as long as you actually use `use-package`. See docs: https://github.com/jwiegley/use-package

**Comment by person808** [comment]
Updated to use use-package.

**Comment by syl20bnr** [comment]
I did not forget you, I will review this PR soon ;-)

**Comment by syl20bnr** [comment]
At last, it is in `develop` :-)
Thank you ! :+1:

**Comment by ralesi** [comment]
@person808,@trishume I had a question about `evil-commentary`.  What value does commentary have over simply defining `evilnc-comment-operator` as below?

````
  (define-key evil-motion-state-map (kbd "gc") 'evilnc-comment-operator)
````

**Comment by person808** [comment]
@ralesi I wasn't aware of that. I'll try it out when I have time. Evil-commentary is much simpler than evil-nerd-commenter and aims to emulate the simplicity of vim-commentary. It also allows this:

> :g/TODO/evil-commentary can be used to toggle comments on all lines that contain the string "TODO".

Can evil-nerd-commenter do that?

**Comment by syl20bnr** [comment]
Maybe we could keep only one of them ?
I'll try vim-commentary.

**Comment by person808** [comment]
@ralesi You are right, it is not different functionality wise. `:g/TODO/evilnc-comment-or-uncomment-lines` is equivalent to `:g/TODO/evil-commentary`. 

@syl20bnr What do you think? Should we remove the layer and add a tip under commenting in the documentation?

**Comment by ralesi** [comment]
I think if anything we should keep evilnc and add functionality to `GC` operator. Is there a performance loss anyone has seen with evil-nc?

**Pull request closed without merging**